tests/integration/test_slurm.py

```diff
@@ -305,3 +305,37 @@ def test_undefined_additional_stores(worker, job_controller) -> None:
         job_controller.count_jobs(states=[JobState.COMPLETED, JobState.REMOTE_ERROR])
         == 2
     )
+
+
+@pytest.mark.parametrize(
+    "worker",
+    ["test_remote_limited_worker"],
+)
+def test_priority(worker, job_controller) -> None:
+    # test only on a limited worker to be sure that only one job will run at the time
+    from jobflow import Flow
+
+    from jobflow_remote import submit_flow
+    from jobflow_remote.jobs.runner import Runner
+    from jobflow_remote.jobs.state import JobState
+    from jobflow_remote.testing import add
+
+    add1 = add(1, 1)
+    flow1 = Flow([add1])
+    submit_flow(flow1, worker=worker, priority=2)
+    add2 = add(1, 1)
+    flow2 = Flow([add2])
+    submit_flow(flow2, worker=worker, priority=1)
+    add3 = add(1, 1)
+    flow3 = Flow([add3])
+    submit_flow(flow3, worker=worker, priority=3)
+
+    runner = Runner()
+    runner.run_all_jobs(max_seconds=120)
+
+    assert job_controller.count_jobs(states=[JobState.COMPLETED]) == 3
+    # check that the jobs were executed according to the priority
+    jobs_info = job_controller.get_jobs_info()
+    jobs_info = sorted(jobs_info, key=lambda x: x.priority, reverse=True)
+    for i in range(len(jobs_info) - 1):
+        assert jobs_info[i].end_time < jobs_info[i + 1].start_time
```

Comment:
With just three jobs, maybe the probability that jobs are executed according to the priority by chance is high ? Currently, it wouldn't be the case because mongo fetches the next job from the latest modified one (I think ?) but I don't think it guarantees that (or does it ?). Maybe consider making e.g. 10 jobs with priorities from 1 to 10 in a predefined random order and check they are executed in the right order. What do you think ?

Reply:
switched to 5 to reduce the likelyhood of random correct order, but to avoid increasing too much the number of job to be executed.